components/director/internal/tenantfetchersvc/handler.go

```diff
@@ -75,6 +78,11 @@ type TenantProviderConfig struct {
 	SubscriptionProviderAppNameProperty string `envconfig:"APP_TENANT_PROVIDER_SUBSCRIPTION_PROVIDER_APP_NAME_PROPERTY,default=subscriptionProviderAppNameProperty"`
 }
 
+// Dependency represents dependency returned from getDependencies callback
```

Comment:
probably you would need another struct / in tenantfetcher just the xsappname is needed